docs/native-modules.md

```diff
@@ -312,21 +320,24 @@ This can be also tied in with C++/WinRT event handlers or `IAsyncOperation<T>` l
 
 ```cpp
     REACT_METHOD(GetString, L"getString");
-    void GetString(bool error, React::ReactPromise<React::JSValue>&& result) noexcept
+    void GetString(React::ReactPromise<std::string>&& result) noexcept
     {
       if (error) {
         result.Reject("Failure");
       } else {
         something.Completed([result] (const auto& operation, const auto& status) {
           // do error checking, e.g. status should be Completed
-          std::wstring wtext{operation.GetResults()};
-          result.Resolve(React::JSValue{ ConvertWideStringToUtf8String(text) });
+          winrt::hstring result{operation.GetResults()};
+          result.Resolve(winrt::to_string(result));
         });
       }
     }
 ```
+See [JavaScript and Windows Runtime strings](#javascript-and-windows-runtime-strings) for more details.
 
-`REACT_SYNC_METHOD` isn't supported in Debug when the JS engine is Chrome V8. You will get an exception that reads:
+The [`JSValue`](native-modules-jsvalue.md) type can be used when the API returns a JavaScript objects or takes JavaScript objects as input parameters.
+
+Native modules will want to use `REACT_METHOD` instead of `REACT_SYNC_METHOD` since the latter precludes web debugging and has performance implications. When using web debugging you will see an exception that reads:

```

Comment:
>performance implications [](start = 127, length = 24)

I do not think it is a true statement. On their own the synchronous methods do not cause perf issues. In future TurboModules all methods are planned to be synchronous according to the FB team. Should we rather speak about the differences and limitations instead of on insisting using the REACT_METHOD?

Reply:
> I do not think it is a true statement. On their own the synchronous methods do not cause perf issues. In future TurboModules all methods are planned to be synchronous according to the FB team

I integrated this from @NickGerleman 's  [PR](https://github.com/microsoft/react-native-windows-samples/pull/271/files#diff-3faf77f987916cbf9f9a99f42e49b9d1R295) but I can take it out if it is wrong.

> Should we rather speak about the differences and limitations instead of on insisting using the REACT_METHOD?

Absolutely, we should explain the differences. I believe you have an issue filed on this: #254
Should I remove this paragraph and look to you to update this soon @vmoroz ?

Reply:
Filed 289